### PR TITLE
Telemetry Adjustments

### PR DESCRIPTION
**Pull request by southpolesteve**
- Enables telemetry for fetch calls
- Removes "LoadingStatus" event from telemetry which was causing a lot of noise
- Add linting to telemetry files

### CODE REVIEW

**Review by notlaforge — APPROVED** [review]

File: src/Shared/Telemetry/TelemetryProcessor.ts [hunk 1]
```diff
@@ -8,123 +8,124 @@ import { userContext } from "../../UserContext";
...
-        actionModifier: ActionModifiers.Start,
-        timestamp: timestamp,
-        data: JSON.stringify(data)
-      }
-    });
-
-    appInsights.startTrackEvent(Action[action]);
-    return timestamp;
-  }
+export function trace(action: Action, actionModifier: string = ActionModifiers.Mark, data?: unknown): void {
```
Comment (languy) [inline src/Shared/Telemetry/TelemetryProcessor.ts:12]: Can we use the fat arrow syntax?
```suggestion
export const trace = (action: Action, actionModifier: string = ActionModifiers.Mark, data?: unknown): void => {
```

File: src/Shared/Telemetry/TelemetryProcessor.ts [hunk 2]
```diff
@@ -8,123 +8,124 @@ import { userContext } from "../../UserContext";
...
+      timestamp: timestamp || Date.now(),
+      data: JSON.stringify(data)
+    }
+  });
+
+  appInsights.stopTrackEvent(Action[action], getData(data));
+}
+
+export function traceOpen(action: Action, data?: unknown, timestamp?: number): number {
+  const validTimestamp = timestamp || Date.now();
```
Comment (languy) [inline src/Shared/Telemetry/TelemetryProcessor.ts:84]: I was going to suggest nullish coalescing here `timestamp && Date.now()`, but a timestamp of 0 is probably not valid, so this is intentional?

**Review by languy — APPROVED** [review]

File: src/Shared/appInsights.ts [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import { ApplicationInsights } from "@microsoft/applicationinsights-web";
 const appInsights = new ApplicationInsights({
   config: {
-    instrumentationKey: "fa645d97-6237-4656-9559-0ee0cb55ee49"
+    instrumentationKey: "fa645d97-6237-4656-9559-0ee0cb55ee49",
+    disableFetchTracking: false
```
[the remark this answers is not in the document]
Comment (southpolesteve) [reply src/Shared/appInsights.ts:5]: Correct. It is disabled by default in app insights. I have no idea why.

[the remark this answers is not in the document]
Comment (southpolesteve) [reply src/Shared/appInsights.ts:5]: Correct. Its disable by default. No idea why they would do that.

File: src/Shared/Telemetry/TelemetryProcessor.ts (hunk 2 above)
Comment (southpolesteve) [reply src/Shared/Telemetry/TelemetryProcessor.ts:84]: I think so. This was existing code, just formatting changes when the class got removed.

**Pull request merged**